overlord/snapstate/snapmgr.go

```diff
@@ -551,6 +556,45 @@ func (m *SnapManager) ensureUbuntuCoreTransition() error {
 	return nil
 }
 
+// ensureSnapConfineApparmor ensures that we have a valid apparmor
+// profile for snap-confine when we are in re-exec mode
+func (m *SnapManager) ensureSnapConfineApparmor() error {
```

Comment:
Question: what will keep us from keeping unbound number of old profiles around?

Perhaps we could generate the profile in /run and load it from there (the actual path is irrelevant as long as it is loaded into the kernel). This doesn't prevent unbound in-memory profiles but at least a "reboot" will fix that.